gitlab/docker-compose.yaml

```diff
@@ -0,0 +1,79 @@
+version: "3.3"
+
+networks:
+  traefik-proxy:
+    name: traefik-proxy
+
+services:
+  redis:
+    restart: always
+    image: redis:latest
+    security_opt:
+      - no-new-privileges:true
+    command:
+      - --loglevel warning
+    volumes:
+      - redis:/var/lib/redis
+  postgresql:
+    image: postgres:14
+    restart: unless-stopped
+    security_opt:
+      - no-new-privileges:true
+    volumes:
+      - postgres:/var/lib/postgresql/data
+    environment:
+      - DB_NAME=${POSTGRES_DB_NAME}
+      - DB_USER=${POSTGRES_USER}
+      - DB_PASS=${POSTGRES_PASS}
+    healthcheck:
+      test:
+        [
+          "CMD-SHELL",
+          "pg_isready -U $$POSTGRES_USER -d $$POSTGRES_DB_NAME"
+        ]
+  gitlab:
+    image: gitlab/gitlab-ce:latest
+    container_name: gitlab
+    restart: always
+    environment:
+      DEBUG: 'true'
+      DB_ADAPTER: postgresql
+      DB_HOST: postgresql
+      DB_PORT: 5432
+      DB_USER: ${POSTGRES_USER}
+      DB_PASS: ${POSTGRES_PASS}
+      DB_NAME: ${POSTGRES_DB_NAME}
+      GITLAB_ROOT_PASSWORD: ${GITLAB_ROOT_PASSWORD}
+      GITLAB_HOST: https://${GITLAB_TRAEFIK_HOST}
+      GITLAB_OMNIBUS_CONFIG: |
+        gitlab_rails['gitlab_shell_ssh_port'] = ${GITLAB_SSH_PORT}
+    networks:
+      - traefik-proxy
+    security_opt:
+      - no-new-privileges:true
+    volumes:
+      - data:/var/lib/gitlab
+      - config:/etc/gitlab
+      - /etc/timezone:/etc/timezone:ro
+      - /etc/localtime:/etc/localtime:ro
+    labels:
+      - "traefik.enable=true"
+      ## Web
+      - "traefik.http.routers.gitlab-web.rule=Host(`${GITLAB_TRAEFIK_HOST}`)"
+      - "traefik.http.routers.gitlab-web.entrypoints=websecure"
+      - "traefik.http.routers.gitlab-web.service=gitlab-web"
+      - "traefik.http.routers.gitlab-web.tls.certresolver=${ACME_CERT_RESOLVER}"
+      - "traefik.http.services.gitlab-web.loadbalancer.server.port=80"
```

Comment:
server.port is not required as long as the image properly specified an `EXPOSE` in its Dockerfile, traefik will autodetect the port.